src/lib.rs

```diff
@@ -1362,4 +1445,82 @@ mod tests {
             .to_value();
         assert_eq!(Value::Number(3.0), value);
     }
+
+    #[cfg_attr(feature = "derive", duktape_fn)]
+    fn test_rust_fn(input: u8) -> String {
+        format!("test {}", input)
+    }
+
+    #[cfg_attr(feature = "derive", duktape_fn)]
```

Comment:
I find it surprising that this produces two items as the proc macro output.  Could the value be transformed into `static test_rust_complex_fn: DukFunction = unsafe { DukFunction::new(1, |...| { ... }) }` or something that is actually still one item?  Could it be expressed using a trait instead?

Reply:
I can make it one item, where the function being annotated is replaced with the derived version. This would prevent the fn being usable in a non-duktape scenario however. Is this desired?

Reply:
I guess you can still `impl<F> Fn for DukFunction<F> where F: Fn` and so on, to make it callable?

Reply:
implementing Fn is unstable. The solution @samsartor came up with, that I implemented, I think is our best bet.

Reply:
SGTM!